metagraph/src/graph/representation/succinct/boss_chunk_construct.cpp

```diff
@@ -388,25 +401,74 @@ void skip_same_suffix(const KMER &el, Decoder &decoder, size_t suf) {
     }
 }
 
+std::pair<std::vector<std::string>, std::string>
+concatenate_chunks(const std::filesystem::path &dir,
+                   const std::vector<std::string> &dummy_sink_names,
+                   const std::vector<std::string> &real_F_W,
+                   BuildCheckpoint *checkpoint) {
+    const uint8_t alphabet_size = KmerExtractor2Bit().alphabet.size();
+
+    std::vector<std::string> real_split_by_W(alphabet_size);
+    std::string dummy_sink_name = dir / "dummy_sink";
+    for (TAlphabet W = 0; W < alphabet_size; ++W) {
+        real_split_by_W[W] = dir/("real_split_by_W_" + std::to_string(W));
+    }
+
+    assert(checkpoint->checkpoint() >= 4);
+    if (checkpoint->checkpoint() > 4) {
+        return { real_split_by_W, dummy_sink_name };
```

Comment:
It would be useful to have a logger message written whenever a step is skipped due to a checkpoint being present, rather than lacking a message.

I see that you did this already in a few places, so maybe you missed this one...